modules/common/src/main/scala/com/snowplowanalytics/snowplow/postgres/streaming/PubSubSink.scala

```diff
@@ -0,0 +1,80 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.postgres.streaming
+
+import scala.concurrent.duration._
+
+import cats.effect.{Async, Resource}
+import cats.implicits._
+
+import com.permutive.pubsub.producer.Model.{ProjectId, Topic}
+import com.permutive.pubsub.producer.encoder.MessageEncoder
+import com.permutive.pubsub.producer.grpc.{GooglePubsubProducer, PubsubProducerConfig}
+
+import org.log4s.getLogger
+
+import com.snowplowanalytics.snowplow.postgres.config.StreamSinkConfig
+
+object PubSubSink {
+
+  private lazy val logger = getLogger
+
+  /**
+    * Set the delay threshold to use for batching. After this amount of time has elapsed (counting
+    * from the first element added), the elements will be wrapped up in a batch and sent.
+    *
+    * If the source MaxQueueSize is small, then the queue could block the app by waiting between
+    * batches if DelayThreshold is too large.  If the source MaxQueueSize is sufficiently large,
+    * then the DelayThreshold should not ever cause blocking.
+    */
+  val DefaultDelayThreshold: FiniteDuration = 200.milliseconds
+
+  /**
+    * A batch of messages will be emitted to PubSub when the batch reaches 1000 messages.
+    * We use 1000 because it is the maximum batch size allowed by PubSub.
+    * This overrides the permutive library default of `5`
+    */
+  val DefaultPubsubMaxBatchSize = 1000L
+
+  /**
+    * A batch of messages will be emitted to PubSub when the batch reaches 10 MB.
+    * We use 10MB because it is the maximum batch size allowed by PubSub.
+    */
+  val DefaultPubsubMaxBatchBytes = 10000000L
+
+  /**
+    * The number of threads used internally by permutive library to process the callback after message delivery.
+    * The callback does very little "work" so we use the minimum number of threads.
+    * This overrides the permutive library default of 3 * num processors
+    */
+  val DefaultNumCallbackExecutors = 1
+
+  implicit val encoder: MessageEncoder[Array[Byte]] = new MessageEncoder[Array[Byte]] {
+    override def encode(message: Array[Byte]): Either[Throwable, Array[Byte]] = message.asRight
+  }
+
+  def create[F[_]: Async](config: StreamSinkConfig.PubSub): Resource[F, StreamSink[F]] = {
+    val producerConfig = PubsubProducerConfig[F](
+      batchSize = DefaultPubsubMaxBatchSize,
```

Comment:
Should these configurable ?

Reply:
they are [configurable](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/sqs2kinesis/sqs2kinesis-configuration-reference/) for sqs2kinesis but I'm not sure if it is a standard for us now cc @istreeter

Reply:
sqs2kinesis was an experiment in configuration!  I certainly liked how I ended up with [all constants defined in the same place](https://github.com/snowplow-incubator/sqs2kinesis/blob/1.0.0/src/main/resources/application.conf).  The benefits were:

- Easier to know where to look for a default value.
- Easier to test optimum settings when benchmarking a rc
- Opportunity to change a setting in production in an emergency.

But it was just an experiment.  I am not strongly opinionated about how to do it here.

Reply:
I am in favour of not making them configurable because I think using them requires really good understanding of internals. However, I am not strongly opinionated neither.